topic_dynamics/modeling.py

```diff
@@ -0,0 +1,326 @@
+"""
+
+import csv
+from operator import itemgetter
+import os
+from typing import List, Tuple
+
+import artm
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+
+def create_batches(directory: str, name: str) -> Tuple[artm.BatchVectorizer, artm.Dictionary]:
+    """
+    Create the batches and the dictionary from the dataset.
+    :param directory: the directory with the dataset.
+    :param name: name of the processed dataset.
+    :return: BatchVectorizer and Dictionary.
+    """
+    print("Creating the batches and the dictionary of the data.")
+    batch_vectorizer = artm.BatchVectorizer(data_path=directory, data_format="bow_uci",
+                                            collection_name=name,
+                                            target_folder=os.path.abspath(os.path.join(directory, name + "_batches")))
+    dictionary = batch_vectorizer.dictionary
+    return batch_vectorizer, dictionary
+
+
+def define_model(n_topics: int, dictionary: artm.Dictionary, sparce_theta: float, sparse_phi: float,
+                 decorrelator_phi: float) -> artm.artm_model.ARTM:
+    """
+    Define the ARTM model.
+    :param n_topics: number of topics.
+    :param dictionary: batch vectorizer dictionary.
+    :param sparce_theta: sparse theta parameter.
+    :param sparse_phi: sparse phi Parameter.
+    :param decorrelator_phi: decorellator phi Parameter.
+    :return: ARTM model.
+    """
+    print("Defining the model.")
+    topic_names = ["topic_{}".format(i) for i in range(1, n_topics + 1)]
+    model_artm = artm.ARTM(topic_names=topic_names, cache_theta=True,
+                           scores=[artm.PerplexityScore(name="PerplexityScore",
+                                                        dictionary=dictionary),
+                                   artm.SparsityPhiScore(name="SparsityPhiScore"),
+                                   artm.SparsityThetaScore(name="SparsityThetaScore"),
+                                   artm.TopicKernelScore(name="TopicKernelScore",
+                                                         probability_mass_threshold=0.3),
+                                   artm.TopTokensScore(name="TopTokensScore", num_tokens=15)],
+                           regularizers=[artm.SmoothSparseThetaRegularizer(name="SparseTheta",
+                                                                           tau=sparce_theta),
+                                         artm.SmoothSparsePhiRegularizer(name="SparsePhi", tau=sparse_phi),
+                                         artm.DecorrelatorPhiRegularizer(name="DecorrelatorPhi", tau=decorrelator_phi)])
+    return model_artm
+
+
+def train_model(model: artm.artm_model.ARTM, n_document_passes: int, n_collection_passes: int,
+                dictionary: artm.Dictionary, batch_vectorizer: artm.BatchVectorizer) -> None:
+    """
+    Train the ARTM model.
+    :param model: the trained model.
+    :param n_document_passes: number of document passes.
```

Comment:
```suggestion
    :param n_doc_iter: number of document passes.
```
`doc` is standard shortening, `iter` is more standard than `passes` (even in bigARTM doc is used as you wrote)